main.go

```diff
@@ -98,6 +139,39 @@ func StartServer(cfg *WebserverConfig, mux http.Handler) error {
 		MaxHeaderBytes: 1 << 20, // 1mb
 	}
 
+	if len(cfg.AutocertDomains) != 0 {
+		log.Printf("Starting server on ports 443 and 80, using autocert domains %v.", cfg.AutocertDomains)
+
+		cache := autocert.DirCache(cfg.AutocertDir)
+		manager := &autocert.Manager{
+			Cache:      cache,
+			Prompt:     autocert.AcceptTOS,
+			HostPolicy: autocert.HostWhitelist(cfg.AutocertDomains...),
+		}
+
+		certHandler := manager.HTTPHandler(http.HandlerFunc(redirectToHTTPS))
+		tlsConfig := manager.TLSConfig()
+		tlsConfig.MinVersion = tls.VersionTLS12
```

Comment:
I believe the `MinVersion` field already defaults to TLSv1.2, at least as of golang v1.22

```go
package tls // import "crypto/tls"

type Config struct {
    // MinVersion contains the minimum TLS version that is acceptable.
    // 
    // By default, TLS 1.2 is currently used as the minimum. TLS 1.0 is the minimum
    // supported by this package.
    // 
    // The server-side default can be reverted to TLS 1.0 by including the value
    // "tls10server=1" in the GODEBUG environment variable.
    MinVersion uint16

    // ... other fields elided ...
}
```